cli/upload/upload.go

```diff
@@ -109,6 +115,28 @@ func runUploadCommand(command *cobra.Command, args []string) {
 	})
 	if err != nil {
 		feedback.Errorf(tr("Error during Upload: %v"), err)
+
+		// Check the error type to give the user better feedback on how
+		// to resolve it
+		var platformErr *arduino.PlatformNotFoundError
+		if errors.As(err, &platformErr) {
+			split := strings.Split(platformErr.Platform, ":")
+			if len(split) < 2 {
+				panic(tr("Platform ID is not correct"))
+			}
+
+			pm := commands.GetPackageManager(instance.GetId())
+			platform := pm.FindPlatform(&packagemanager.PlatformReference{
+				Package:              split[0],
+				PlatformArchitecture: split[1],
+			})
+
+			if platform != nil {
+				feedback.Errorf(tr("Try running `%s core install %s`", globals.VersionInfo.Application, platformErr.Platform))
+			} else {
+				feedback.Errorf(tr("Platform %s is not found in any known index", platformErr.Platform))
+			}
+		}
```

Comment:
the code is the same as in `compile.go`. But I would't know where to put it

Reply:
We may just always print `Try running './arduinoc-li core install arduino:samd'` and let the `core install` command print that the `Platform %s is not found in any known index` if appropriate.

Reply:
I wouldn't do that, we're giving the user a solution that won't work. We're already know what's going to happen if she runs that, might has already say it.